src/app/CommandHandler.h

```diff
@@ -416,21 +507,31 @@ class CommandHandler : public Messaging::ExchangeDelegate
      */
     void SetGroupRequest(bool isGroupRequest) { mGroupRequest = isGroupRequest; }
 
+    CommandPathRegistry & GetCommandPathRegistry() const { return *mCommandPathRegistry; }
+
+    size_t MaxPathsPerInvoke() const { return mMaxPathsPerInvoke; }
+
     Messaging::ExchangeHolder mExchangeCtx;
     Callback * mpCallback = nullptr;
     InvokeResponseMessage::Builder mInvokeResponseBuilder;
     TLV::TLVType mDataElementContainerType = TLV::kTLVType_NotSpecified;
     size_t mPendingWork                    = 0;
-    bool mSuppressResponse                 = false;
-    bool mTimedRequest                     = false;
 
-    bool mSentStatusResponse = false;
-
-    State mState       = State::Idle;
-    bool mGroupRequest = false;
     chip::System::PacketBufferTLVWriter mCommandMessageWriter;
     TLV::TLVWriter mBackupWriter;
-    bool mBufferAllocated = false;
+    size_t mMaxPathsPerInvoke = CHIP_CONFIG_MAX_PATHS_PER_INVOKE;
+    // TODO Allow flexibility in registration.
+    BasicCommandPathRegistry<CHIP_CONFIG_MAX_PATHS_PER_INVOKE> mBasicCommandPathRegistry;
+    CommandPathRegistry * mCommandPathRegistry = &mBasicCommandPathRegistry;
```

Comment:
This adds ~100 bytes of RAM in the default configuration, right?  Please make sure a followup is filed to look into reducing the memory impact.

Reply:
Added TODO comment and added it to the list of TODO on https://github.com/project-chip/connectedhomeip/issues/30453